src/FactoryCollection.php

```diff
@@ -18,12 +18,18 @@ final class FactoryCollection
     /** @var int */
     private $max;
 
+    /** @var iterable<array<string, mixed>>|null */
+    private $sequence;
+
     /**
-     * @param int|null $max If set, when created, the collection will be a random size between $min and $max
+     * @param int|null                       $max      If set, when created, the collection will be a random size between $min and $max
+     * @param iterable<array<string, mixed>> $sequence
      *
      * @psalm-param Factory<TObject> $factory
+     *
+     * @deprecated using directly FactoryCollection's constructor is deprecated. It will be private in v2. Use named constructors instead.
      */
-    public function __construct(Factory $factory, int $min, ?int $max = null)
+    public function __construct(Factory $factory, ?int $min = null, ?int $max = null, ?iterable $sequence = null)
```

Comment:
```suggestion
    public function __construct(Factory $factory, ?int $min = null, ?int $max = null, ?iterable $sequence = null, bool $calledInternally = false)
```

Then trigger a deprecation if `$calledInternally = false`

Reply:
ok, that's clever, I was actually thinking the annotation was enough